modules/services/swaync.nix

```diff
@@ -0,0 +1,154 @@
+{ pkgs, lib, config, ... }:
+let
+  cfg = config.services.swaync;
+
+  jsonFormat = pkgs.formats.json { };
+
+  settings = {
+    "$schema" = "${cfg.package}/etc/xdg/swaync/configSchema.json";
+  } // cfg.settings;
+
+  configFile = pkgs.writeTextFile {
+    name = "swaync/config.json";
+    text = builtins.toJSON settings;
+    # TODO uncomment once https://github.com/ErikReider/SwayNotificationCenter/issues/298 fixed
+    # checkPhase = "${pkgs.check-jsonschema}/bin/check-jsonschema --schemafile ${settings."$schema"} $out ";
+  };
+in {
+  meta.maintainers = [ lib.maintainers.rhoriguchi ];
+
+  options.services.swaync = {
+    enable = lib.mkEnableOption "Swaync notification daemon";
+
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = pkgs.swaynotificationcenter;
+      defaultText = lib.literalExpression "pkgs.swaynotificationcenter";
+      description = ''
+        Swaync package to use. Set to <code>null</code> to use the default package.
+      '';
+    };
+
+    systemd = {
+      enable = lib.mkEnableOption "Systemd integration";
+
+      target = lib.mkOption {
+        type = lib.types.str;
+        default = "graphical-session.target";
+        example = "sway-session.target";
+        description = ''
+          Systemd target to bind to.
+        '';
+      };
+    };
+
```

Comment:
I would suggest always creating the systemd service.
```suggestion
```

Reply:
My thought process was, if you only want to use the notification shade but not the notification popups. This way you can configure it without starting it.

Reply:
One could still disable the systemd in its own config via systemd.user.services.swaync.enable = false or something similar (not sure how to disable the service) , i.e., it makes the module simpler.